File: MoutsTI.Infra/Context/Employee.cs
namespace MoutsTI.Infra.Context;

public partial class Employee
{
    public long EmployeeId { get; set; }

    public long RoleId { get; set; }

    public long? ManagerId { get; set; }

    public string FirstName { get; set; } = null!;

    public string LastName { get; set; } = null!;

    public string Email { get; set; } = null!;

    public string DocNumber { get; set; } = null!;

    public string Password { get; set; } = null!;

    public DateTime Birthday { get; set; }

    public virtual ICollection<EmployeePhone> EmployeePhones { get; set; } = new List<EmployeePhone>();

    public virtual ICollection<Employee> InverseManager { get; set; } = new List<Employee>();

    public virtual Employee? Manager { get; set; }

    public virtual EmployeeRole Role { get; set; } = null!;
}

File: moutsti.presentation/src/types/employee.ts
export interface EmployeeRole {
  roleId: number;
  name: string;
}

export interface Manager {
  employeeId: string;
  firstName: string;
  lastName: string;
  fullName: string;
  email: string;
}

export interface Employee {
  employeeId: number;
  firstName: string;
  lastName: string;
  fullName: string;
  email: string;
  docNumber?: string;
  birthday?: string;
  roleId: number;
  role: EmployeeRole;
  managerId?: number;
  manager?: Manager;
  phones: string[];
}

export interface CreateEmployeeDTO {
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  docNumber?: string;
  birthday?: string;
  roleId: number;
  managerId?: number;
  phones: string[];
}

export interface UpdateEmployeeDTO extends CreateEmployeeDTO {
}

export interface LoginRequest {
  email: string;
  password: string;
}

export interface LoginResponse {
  token: string;
  expiresAt: string;
  employee: Employee
}

export interface ApiError {
  message: string;
  statusCode: number;
}
